createdisk.sh

```diff
@@ -15,36 +15,57 @@ INSTALL_DIR=${1:-crc-tmp-install-data}
 
 # If the user set OKD_VERSION in the environment, then use it to set BASE_OS
 OKD_VERSION=${OKD_VERSION:-none}
-destDirPrefix="crc"
 if [[ ${OKD_VERSION} != "none" ]]
 then
     BASE_OS=fedora-coreos
     destDirPrefix="crc_okd"
 fi
 BASE_OS=${BASE_OS:-rhcos}
 OPENSHIFT_VERSION=$(${JQ} -r .clusterInfo.openshiftVersion $INSTALL_DIR/crc-bundle-info.json)
+BUNDLE_TYPE=$(${JQ} -r .type $INSTALL_DIR/crc-bundle-info.json)
+destDirPrefix="crc"
+if [ ${BUNDLE_TYPE} == "microshift" ]; then
+    destDirPrefix="crc_${BUNDLE_TYPE}"
+    BASE_OS=rhel
+fi
 
 # SNC_PRODUCT_NAME: If user want to use other than default product name (crc)
 # VM_PREFIX: short VM name (set by SNC_PRODUCT_NAME) + random string generated by openshift-installer
 SNC_PRODUCT_NAME=${SNC_PRODUCT_NAME:-crc}
-VM_PREFIX=$(get_vm_prefix ${SNC_PRODUCT_NAME})
-VM_NAME="${VM_PREFIX}-master-0"
+if [ ${BUNDLE_TYPE} == "microshift" ]; then
+    VM_NAME=${SNC_PRODUCT_NAME}
+else
+    VM_PREFIX=$(get_vm_prefix ${SNC_PRODUCT_NAME})
+    VM_NAME="${VM_PREFIX}-master-0"
+fi
 
-# Get the IP of the VM
 VM_IP=$(sudo virsh domifaddr ${VM_NAME} | tail -2 | head -1 | awk '{print $4}' | cut -d/ -f1)
 
-# Remove unused images from container storage
-${SSH} core@${VM_IP} -- 'sudo crictl rmi --prune'
-
-# Disable kubelet service
-${SSH} core@${VM_IP} -- sudo systemctl disable kubelet
-
-# Stop the kubelet service so it will not reprovision the pods
-${SSH} core@${VM_IP} -- sudo systemctl stop kubelet
+if [ ${BUNDLE_TYPE} != "microshift" ]; then
+    # Remove unused images from container storage
+    ${SSH} core@${VM_IP} -- 'sudo crictl rmi --prune'
+    
+    # Disable kubelet service
+    ${SSH} core@${VM_IP} -- sudo systemctl disable kubelet
+    
+    # Stop the kubelet service so it will not reprovision the pods
+    ${SSH} core@${VM_IP} -- sudo systemctl stop kubelet
+fi
 
 # Enable the podman.socket service for API V2
 ${SSH} core@${VM_IP} -- sudo systemctl enable podman.socket
 
+if [ ${BUNDLE_TYPE} == "microshift" ]; then
+    # Pull openshift release images
+    ${SSH} core@${VM_IP} 'sudo bash -x -s' <<EOF
+     jq --raw-output '.images | to_entries | map(.value) | join("\n")' /usr/share/microshift/release/release-$(uname -i).json | xargs -n1 podman pull --authfile /etc/crio/openshift-pull-secret
+EOF
+    # Extend the lvm to use all free space
+    ${SSH} core@${VM_IP} -- sudo lvextend -r -l +100%FREE /dev/mapper/rhel-root
+    # Disable firewalld
+    ${SSH} core@${VM_IP} -- sudo systemctl disable firewalld
```

Comment:
Why? Since you added a comment before the command, it's a good opportunity to record why we don't want firewalld.

Reply:
Can you move this to the commit adding the podman changes then?

Reply:
I think you updated the comment to explain why it's not podman specific?